### PR TITLE
Add provider `lunatic`

### PR DESCRIPTION
**Pull request by pinkforest**
This PR adds [lunatic](https://github.com/lunatic-solutions/lunatic) provider

I propose to either or / all of these:
- [X] Change the private fn from RustProvider as `pub(crate)` for DRY at lunatic that wraps them over and extends
- [X] Add provider that wraps on top of Rust since lunatic-rust is the 1st class server side wasm citizen
- [X] Detect from .cargo/config.toml whether the runner is set as "lunatic"
- [X] Detect whether the target is wasm32-wasi

Questions
- [X] Lunatic needs [lunatic-runtime](https://crates.io/crates/lunatic) binary and I was wondering how could I cache this somewhere best suited to nixpacks - **can use it's nixpkg - it was recently bumped to 0.13**

In the future
- Should this use alpine instead of big ubuntu ? musl shouldn't be an issue
- Extend with AssemblyScript-lunatic and re-organise `providers/lunatic.rs` with `lunatic/rust.rs` and `lunatic/as.rs`
- Add test workspace snapshot e.g. https://github.com/yaws-rs/yaws

Lunatic is fun - a bloggy blog I wrote:
- https://missmissm.medium.com/play-ping-pong-with-lunatic-udp-ef557a22a604
- https://github.com/lunatic-solutions/submillisecond/ - Web framework w/ [submillisecond-liveview](https://github.com/lunatic-solutions/submillisecond-live-view)
- https://github.com/yaws-rs - a web server I'm using it in

### CODE REVIEW

**Review by aleksrutins — APPROVED** [review]
Nice, and it seems to work, too! Just a grammar nitpick on the docs.

File: docs/pages/docs/providers/lunatic.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+```
+cargo build --release
+```
+
+## Start
+
+The binaries get .wasm suffix and will be ran with the lunatic runtime.
+
+If your project has multiple binaries, you can specify which one to run with the `NIXPACKS_RUST_BIN` environment variable.
+Optionally, it can override with the `default_run` property in `Cargo.toml` under the `[package]` section.
```
[the remark this answers is not in the document]
Comment (pinkforest) [reply docs/pages/docs/providers/lunatic.md]: Thanks :)

File: docs/sidebar.ts [hunk 2]
```diff
@@ -44,6 +44,7 @@ export const sidebarItems: ISidebarSection[] = [
       { href: "/docs/providers/go", text: "Go" },
       { href: "/docs/providers/haskell", text: "Haskell" },
       { href: "/docs/providers/java", text: "Java" },
+      { href: "/docs/providers/lunatic", "text: "Lunatic" },
```
Comment (half0wl) [inline docs/sidebar.ts:47]:
```suggestion
      { href: "/docs/providers/lunatic", text: "Lunatic" },
```

Caught by CI in https://github.com/railwayapp/nixpacks/actions/runs/4921867678/jobs/8824973626?pr=873.

**Review by half0wl — APPROVED** [review]
@pinkforest Sorry for the delay here!

This looks good to me. I'll defer to @coffee-cup (lead maintainer of nixpacks) for the final stamp.

File: docs/sidebar.ts (hunk 2 above)
Comment (pinkforest) [reply docs/sidebar.ts]: Thanks good catch :)

**Review by coffee-cup — APPROVED** [review]

File: docs/pages/docs/providers/lunatic.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+---
+title: Lunatic
+---
+
+# {% $markdoc.frontmatter.title %}
+
+Lunatic is detected if both
```
[the remark this answers is not in the document]
Comment (pinkforest) [reply docs/pages/docs/providers/lunatic.md]: Thanks added - have couple of templates and blog write-ups cooking in the oven :partying_face:

**Pull request merged**